strikt-core/src/main/kotlin/strikt/api/Assertion.kt

```diff
@@ -181,7 +183,15 @@ interface Assertion {
           "value of ${function.propertyName}",
           function
         )
-        else -> map("%s", function)
+        else -> {
+          val fieldName = try {
+            val line = FilePeek.getCallerFileInfo().line
+            ParsedMapInstruction(line).body.substringAfter("it.")
```

Comment:
Would this still work if the lambda parameter had an explicit name?

Reply:
no. in that case the block would fail and the default would trigger.

Reply:
actually i tried it and if i change it to 
`          map { sub->sub.birthDate.year }.isEqualTo(1971)`
the error looks like this:
`sub->sub.birthDate.year:`

Reply:
it just take the whole body of the map block, which is useful most of the time

Reply:
That's not bad, yeah